nd2reader/parser.py

```diff
@@ -247,6 +247,28 @@ def _channel_offset(self):
         """
         return {channel: n for n, channel in enumerate(self.metadata["channels"])}
 
+    def _check_unwanted_bytes(self, image_group_data, image_data_start, height, width):
+        number_of_true_channels = int(len(image_group_data[4:]) / (height * width))
+        n_unwanted_bytes = (len(image_group_data[image_data_start:]))%(height*width)
+        if not n_unwanted_bytes:
+            return False
+        assert 0 == n_unwanted_bytes % height, "Unexpected unwanted bytes distribution."
+        byte_ids = range(image_data_start+height*number_of_true_channels, len(image_group_data)-n_unwanted_bytes+1, height*number_of_true_channels)
+        all_zero_bytes = all([0 == image_group_data[byte_ids[i]+i] for i in range(len(byte_ids))])
+        if not all_zero_bytes:
+            warnings.warn(f'Identified {n_unwanted_bytes} unwanted non-zero bytes in the ND2 file.')
+        return all_zero_bytes
+
+    def _remove_unwanted_bytes(self, image_group_data, image_data_start, height, width):
+        # Remove unwanted 0-bytes that can appear in stitched images
```

Comment:
Isn't this function replicating a lot of the logic of `_check_unwanted_bytes`? Maybe you can return the location of the bytes that can be removed from that function and pass it to this one to avoid the code duplication.